Security upgrade guide 7.17-8.x (#5186)
elastic/security-docs: docs/upgrade/upgrade-7.17-8.x.asciidoc

* Review release notes, deprecations, and breaking changes for {security-guide}/release-notes.html[{elastic-sec}], {ref}/es-release-notes.html[{es}], {kibana-ref}/release-notes.html[{kib}], and, if applicable, {fleet-guide}/release-notes.html[{fleet} and {agent}], {beats-ref}/release-notes.html[{beats}], and {logstash-ref}/releasenotes.html[{ls}]. Identify any issues that might affect your deployment. Work with your Support Engineer on any questions you may have. Start with breaking changes for solution and platform components you use, such as {es} and {kib}. 

* Schedule a system maintenance window within your organization.

[float]
=== Pre-upgrade steps

To prepare for the upgrade process, follow these steps before you start:

. Do a software version inventory across your entire Elastic platform, including {es}, {kib}, {agent}, {beats}, and {ls}. If you're managing your deployments using {ece-ref}/ece-upgrade.html[{ece}] (ECE) or {eck-ref}/k8s-upgrading-eck.html[{eck}] (ECK), you may need to upgrade the orchestrator first.

> . If you're running any of these at versions earlier than 7.17, you must first upgrade them all to the latest 7.17.x patch release before upgrading to 8.x. This enables you to use the {kib} Upgrade Assistant to upgrade to 8.x. 

Reviewer: Added the ECK/ECE info as a `Note` after Step 2. Hopefully it fits better here. 
```
NOTE: If you're managing your deployments using {ece-ref}/ece-upgrade.html[{ece}] (ECE) or {eck-ref}/k8s-upgrading-eck.html[{eck}] (ECK), you may need to upgrade the system clusters first.
```

--

. If you're not using {ref}/snapshots-take-snapshot.html#automate-snapshots-slm[{slm} (SLM)], you must set up, configure, and run at least one {ref}/snapshot-restore.html[snapshot]. Rollbacks require a snapshot and cause data loss without one. Snapshots are {ref}/snapshot-restore.html#how-snapshots-work[incremental] – depending on the cluster size and the input/output rate, the initial snapshot may take several hours to complete. If you are using SLM, {ref}/snapshots-take-snapshot.html#check-slm-history[check the SLM history] to ensure that snapshots are completing successfully.

[float]
=== Perform 8.x upgrade on Non-Prod (Optional) or Prod Deployment

NOTE: We strongly recommend you perform the following steps on a non-prod test deployment first to address any potential issues before upgrading on a prod deployment. 

. Back up all of your data to a {ref}/snapshots-take-snapshot.html[snapshot] – which is a backup of indices taken from a running cluster. 

. We recommend you <<rules-api-export, export all your custom detection rules>> in the event there are issues with the detection engine after the upgrade.

> . Upgrade {es}. Refer to {stack-ref}/upgrading-elasticsearch.html[these upgrade instructions]. 

Reviewer: Removed the {es} ref in the top link and moved it to the self-orchestrated item.

** If you're using {ecloud}, we recommend upgrades with no downtime. Refer to {cloud}/ec-upgrade-deployment.html[these instructions].  
** If you're using {ece} (ECE), refer to {ece-ref}/ece-upgrade-deployment.html[these instructions].  
** If you're using {eck} (ECK), refer to {eck-ref}/k8s-upgrading-stack.html[these instructions]. 
** If you're upgrading a self-orchestrated deployment, upgrade the data nodes tier-by-tier, in this order:
... Frozen tier
... Cold tier 
... Warm tier
... Hot tier 
... Any other nodes not in a tier
... All remaining nodes that are neither master-eligible nor data nodes
... Master-eligible nodes

. Upgrade {kib}. Refer to {stack-ref}/upgrading-kibana.html[these instructions].
+

> NOTE: If you're using {ecloud} or {ece}, this is already included in the {es} upgrade.

Reviewer: @shainaraskas not sure what ECH refers to? But I can convert the `{ecloud}` variable to `Elastic Cloud Hosted` if that's compliant w/ terminology standards.

> NOTE: If you're using {ecloud} or {ece}, this is already included in the {es} upgrade.

Reviewer 2: ECH is Elastic Cloud on Heroku. 

The meaning is important here because elastic cloud hosted (fka ESS / elastic cloud service) doesn't cover elastic cloud on kubernetes or elastic cloud on heroku, but those all fall under the umbrella of "Elastic Cloud" (so does Elastic Cloud Enterprise [ECE]).

> NOTE: If you're using {ecloud} or {ece}, this is already included in the {es} upgrade.

Reviewer: I'm learning all kind of gems! Thanks for the info, updated to `Elastic Cloud Hosted`.

. Validate {es} and {kib} are operating as expected by completing the following checks: 
.. {es}
... Check the status of your clusters and ensure they're green. Run the `GET _cat/health` API request. For more information, refer to the {ref}/cat-health.html[cat health API documentation].
... Check the size of your rolled over indices/shards, which should be less than 50 GB. Run the `GET _cat/shards?s=store:desc` API request. For more information, refer to the {ref}/cat-shards.html[cat shards API documentation].    

> ... Ensure the index and search rate are close to what they were prior to upgrading. Go to **Stack Monitoring** -> **ES** -> **Overview**.

Reviewer: Thanks @shainaraskas for the tip. Is this the best doc to ref for stack monitoring setup? https://www.elastic.co/guide/en/kibana/current/configuring-monitoring.html. I also found this [Logstash reference](https://www.elastic.co/guide/en/logstash-kubernetes/current/ls-k8s-stack-monitoring.html) too, but wasn't sure which one was more accurate. I'm assuming the former.

> ... Ensure the index and search rate are close to what they were prior to upgrading. Go to **Stack Monitoring** -> **ES** -> **Overview**.

Reviewer 2: Suggest this one instead: https://www.elastic.co/guide/en/kibana/current/xpack-monitoring.html

+
TIP: You can also check the index doc count using the {ref}/cat-indices.html[cat index API].
... Verify the {slm} (SLM) is taking snapshots by {ref}/snapshots-take-snapshot.html#check-slm-history[checking the SLM history]. 

> ... Ensure {ml} is up and running. 

Reviewer: Revised this statement in the meantime: 
`If you use {ml}, ensure that it is up and running. `

[[upgrade-7.17-8x]]
== Upgrade from 7.17 to an 8.x version

[float]
=== Why it's important to upgrade

Upgrading provides you access to {elastic-sec}'s latest features, enhancements, and bug fixes, many of which enable you to save your company money, respond faster to potential threats, investigate, and analyze data with enhanced tooling. For more benefits, check out our blog, https://www.elastic.co/blog/top-5-reasons-to-upgrade-elastic-security[Top 5 reasons to upgrade Elastic Security]. Also, it's important to ensure your deployment is fully maintained and supported. For more information, refer to https://www.elastic.co/support/eol[Elastic's Product End of Life Dates]. 

[float]
=== Plan for your upgrade

Before upgrading to {elastic-sec} 8.x, consider the following recommendations:

* Plan for an appropriate amount of time to complete the upgrade. Depending on your configuration and the size of your cluster, the process can take up to a week to complete.

> * Open a https://support.elastic.co[support case] with Elastic to alert our Customer Support team of your system change. You can also get support by purchasing a https://www.elastic.co/consulting[consulting engagement]. 

Reviewer 3: This should be mentioned only if the customer has a subscription which grants consultative support.

Reviewer 4: ++ Agreed.  Perhaps "If your support subscription includes consultative support..."

* Choose your desired {elastic-sec} 8.x version to upgrade to. We recommend upgrading to the latest minor and patch version, as this provides the most extensive set of features, bug fixes, and security patches. Be sure to use the same version for upgrading your development or non-production deployment as for your production deployment. 

* Review the latest 8.x {elastic-sec} features and the current set of Elastic connectors, integrations, and detection rules to determine if you can replace any customized content with out-of-the-box functionality. This can help reduce the workload and complexity of your upgrade.

> * If your organization uses external SOAR (security orchestration, automation, and response) workflows on {elastic-sec} alerts (signals), you may need to change your workflows to accommodate the new <<alert-schema, alert schema>> used in 8.x. These changes may take up to two weeks to complete.

Reviewer 4: Agreed.  Let's remove the specific mention of two weeks altogether.  The point here is to inform the upgrader that this process could take some time, as typically it involves changing scripts or code and then testing it to ensure that everything is still working.

> * Review release notes, deprecations, and breaking changes for {elastic-sec}, {es}, {kib}, and, if applicable, {fleet}, {agent}, {beats}, and {ls}. Identify any issues that might affect your deployment. Work with your Support Engineer on any questions you may have. Start with breaking changes for solution and platform components you use, such as {es} and {kib}. 

Reviewer: Yes! Good idea.

. If you're running any of these at versions earlier than 7.17, you must first upgrade them all to the latest 7.17.x patch release before upgrading to 8.x. This enables you to use the {kib} Upgrade Assistant to upgrade to 8.x. 

. Run the {kib-ref}/upgrade-assistant.html[Upgrade Assistant]. Take note of any critical error messages, then work with your Support rep to resolve them.
+
.Requirements
[sidebar]
--
To run the Upgrade Assistant, you must have `superuser` cluster admin permissions.
--

. If you're not using {ref}/snapshots-take-snapshot.html#automate-snapshots-slm[{slm} (SLM)], you must set up, configure, and run at least one {ref}/snapshot-restore.html[snapshot]. Snapshots are {ref}/snapshot-restore.html#how-snapshots-work[incremental] – depending on the cluster size and the input/output rate, the initial snapshot may take several hours to complete. If you are using SLM, {ref}/snapshots-take-snapshot.html#check-slm-history[check the SLM history] to ensure that snapshots are completing successfully.

[float]

> === Perform 8.x upgrade on Non-Prod (Optional) or Prod Deployment

Reviewer 4: This is good, but do we need a similar at the end like:  "Repeat this process for your Prod deployment" (If you did the above for your non-prod/dev deployments)?

Also, where do we put the comment about "If you're using a Cross-cluster search environment, make sure to upgrade your central/remote??? deployments first."  ??  Then add a "Repeat this process for other deployments, etc.

Reviewer: Hi @MikePaquette, added these [additions](https://security-docs_bk_5186.docs-preview.app.elstc.co/guide/en/security/master/upgrade-7.17-8x.html#_perform_8_x_upgrade_on_a_non_prod_optional_or_prod_deployment): 
![2024-05-23_11-40-44](https://github.com/elastic/security-docs/assets/57149392/ca0bcec2-e0dc-40a8-b9c9-b244689b2a07)

and towards the end of the non-prod instructions. 
![2024-05-23_11-41-59](https://github.com/elastic/security-docs/assets/57149392/3fcebbc1-0ded-4961-8f99-3565bd430bf2)

Please let me know if these are correct, thanks!

NOTE: We strongly recommend you perform the following steps on a non-prod test deployment first to address any potential issues before upgrading on a prod deployment. 

> . Back up all of your data to a {ref}/snapshots-take-snapshot.html[snapshot] – which is a backup of an index taken from a running cluster. Rollbacks require a snapshot and cause data loss without one.

Reviewer 4: Question: Should "backup of an index" be "back of indices" ?

. We recommend you <<rules-api-export, export all your custom detection rules>> in the event there are issues with the detection engine after the upgrade.

. Upgrade {es}. Refer to {stack-ref}/upgrading-elasticsearch.html[these upgrade instructions]. 

> ** If you're using {ecloud}, we recommend upgrades with no downtime. Refer to {cloud}/ec-upgrade-deployment.html[these instructions].  

Reviewer 5: ```suggestion
** If you're using {ecloud}, refer to {cloud}/ec-upgrade-deployment.html[these instructions].  
```
Ignore this suggestion if the part I deleted is important, but this would be more aligned with the format of the following bullets.

Reviewer: Sure - We can eliminate this I think. Thanks.

** If you're using {ece} (ECE), refer to {ece-ref}/ece-upgrade-deployment.html[these instructions].  
** If you're using {eck} (ECK), refer to {eck-ref}/k8s-upgrading-stack.html[these instructions]. 

> ** If you're upgrading a self-orchestrated deployment, upgrade the data nodes first, tier-by-tier, in this order:

Reviewer 5: Not clear what "first" means here. Before what? If its just before proceeding to step 4, I don't think we need to include it.

... Frozen tier
... Cold tier 
... Warm tier
... Hot tier 
... Any other nodes not in a tier
... All remaining nodes that are neither master-eligible nor data nodes
... Master-eligible nodes

. Upgrade {kib}. Refer to {stack-ref}/upgrading-kibana.html[these instructions].
+
NOTE: If you're using {ecloud} or {ece}, this is already included in the {es} upgrade.

. Validate {es} and {kib} are operating as expected by completing the following checks: 
.. {es}
... Check the status of your clusters and ensure they're green. Run the `GET _cat/health` API request. For more information, refer to the {ref}/cat-health.html[cat health API documentation].

> ... Check the size of your rolled over indices/shards, which should be less than 50 GB. Run the `GET _cat/shards?s=store:desc` API request. For more information, refer to the {ref}/cat-shards.html[cat shards API documentation].    

Reviewer 5: Not clear about the 50GB mention here. Is this something users need to ensure at this time in the process? What if they have shards that are larger than 50GB? (Google tells me there's no hard limit on shard size.)

Reviewer: Going to remove this statement about checking the size, per @shainaraskas' recommendation.

... Ensure the index and search rate are close to what they were prior to upgrading. Go to **Stack Monitoring** -> **ES** -> **Overview**.
+
TIP: You can also check the index doc count using the {ref}/cat-indices.html[cat index API].
... Verify the {slm} (SLM) is taking snapshots by {ref}/snapshots-take-snapshot.html#check-slm-history[checking the SLM history]. 
... Ensure {ml} is up and running. 
.. {kib} 
... Ensure that you and your users can successfully log in to {kib} and access desired pages.
... Check {kibana-ref}/discover.html[Discover] and verify that the index patterns you normally use are available.
... Verify that your commonly-used {kibana-ref}/dashboard.html[dashboards] are available and working properly.

> ... Check your SIEM detection rules, and ensure that the desired set of rules is enabled (rule management) and that enabled rules are running without errors or warnings (rule monitoring).

Reviewer 4: Is this item redundant with the similar item below under Validate that Elastic Security is operating OK?

... If you use any watcher-based {kib} scheduled {kibana-ref}/reporting-getting-started.html[reporting], ensure that it is working properly.

. Upgrade your ingest components (such as {ls}, {fleet} and {agent}, {beats}, etc.) Refer to the {stack-ref}/upgrading-elastic-stack.html[Elastic Stack upgrade docs] for details.

. Validate Ingest is operating OK.
.. Open *Discover*, go through data views for each of your expected ingest data streams, and ensure that data is being ingested in the expected format and volume. 

. Validate that {elastic-sec} is operating OK.
.. Ensure that your detection rules are re-enabled.
.. Ensure that any SOAR workflows that consume alerts are working.
.. Verify that any custom dashboards your team has created are working properly, especially if they operate on alert (signal) documents.

. Confirm with your appropriate stakeholders that the upgrade process is complete.

[float]
=== Post-upgrade steps

The following sections describe procedures to complete after upgrading {elastic-sec} to 8.x.

[float]
[[reenable-rules-upgrade]]
==== Re-enable disabled rules

> Any active rules when you upgrade from 7.17 to 8.0.1 or newer are automatically disabled, and a tag named `auto_disabled_8.0` is added to those rules for tracking purposes. Once the upgrade is complete, you can filter rules by the newly added tag, then use bulk actions to re-enable them:

Reviewer 5: ```suggestion
Any active rules when you upgrade from 7.17 to 8.0.1 or newer are automatically disabled, and a tag named `auto_disabled_8.0` is added to those rules for tracking purposes. Once the upgrade is complete, you can filter rules by the new tag, then use bulk actions to re-enable them:
```

. Go to the Rules page (*Detect -> Rules*).
. From the *Tags* dropdown, search for `auto_disabled_8.0`.
. Click *Select all _x_ rules*, or individually select the rules you want to re-enable.
. Click *Bulk actions -> Enable* to re-enable the rules.

Alternatively, you can use the <<bulk-actions-rules-api, Bulk rule actions>> API to re-enable rules.

[float]
[[fda-upgrade]]
==== Full Disk Access (FDA) approval for {elastic-endpoint}

When you manually install {elastic-endpoint}, you must approve a system extension, kernel extension, and enable Full Disk Access (FDA). There is a new FDA requirement in 8.x. Refer to <<elastic-endpoint-deploy-reqs>> to review the required permissions.

[float]
[[data-views-upgrade]]
==== Requirements to display Data views in the {security-app}

To make the *Data view* option appear in an environment with legacy alerts, a user with elevated role privileges must visit the {security-app}, open a page that displays alert data (such as the Overview page), then refresh the page. The user's role privileges must allow them to enable the detections feature in a {kib} space. Refer to <<enable-detections-ui, Enable and access detections>> for more information.

> NOTE: If new alerts are generated in an upgraded environment without legacy alerts, refreshing any page with alert data in {elastic-sec} will make the *Data view* option appear in the {elastic-sec} UI.

Reviewer 5: This note was a bit confusing to me, I think you could remove it and change the parenthetical in the previous paragraph to say "...open a page that displays alert data (at least one alert must be present), then refresh the page." The current parenthetical that I'm suggesting we replace, "(such as the Overview page)" could probably be safely replaced since the Alerts page is so prominent in the nav menu.

[float]
[[alert-schema-upgrade]]
==== New alert schema

The system index for detection alerts has been renamed from `.siem-signals-<space-id>` to `.alerts-security.alerts-<space-id>` and is now a hidden index. Therefore, the schema used for alert documents in {elastic-sec} has changed. Users that access documents in the `.siem-signals` indices using the {elastic-sec} API must modify their API queries and scripts to operate properly on the new 8.x alert documents. Refer to <<query-alert-indices, how to query alert indices>> and review the new <<alert-schema, Alert schema>>.

[float]
[[preview-upgrade]]
==== New privileges required to view alerts and preview rules

* To view alerts, users need `manage`, `write`, `read`, and `view_index_metadata` privileges to two new indices, `.alerts-security.alerts` and `.internal.alerts-security.alerts`. Existing users who are upgrading to 8.x can retain their privileges to the `.siem-signals` index.

* To <<preview-rules, preview rules>>, users need `read` access to the new `.preview.alerts-security.alerts` index. Refer to <<detections-permissions-section>> for more information.

[float]
[[im-rules-upgrade]]
==== Updates to indictor match rules

> Changes to the indicator match rule's <<rule-ui-advanced-params, default threat indicator path>> might require you to update existing rules or create new ones after upgrading to 8.x. Be mindful of the following:

Reviewer 5: ```suggestion
Changes to the indicator match rule type's <<rule-ui-advanced-params, default threat indicator path>> might require you to update existing rules or create new ones after upgrading to 8.x. Be mindful of the following:
```

[[upgrade-7.17-8x]]
== Upgrade from 7.17 to an 8.x version

[float]
=== Why it's important to upgrade

Upgrading provides you access to {elastic-sec}'s latest features, enhancements, and bug fixes, many of which enable you to save your company money, respond faster to potential threats, and improve the tools you use to investigate and analyze your data. For more benefits, check out our blog, https://www.elastic.co/blog/top-5-reasons-to-upgrade-elastic-security[Top 5 reasons to upgrade Elastic Security]. Also, it's important to ensure that your deployment is fully maintained and supported. For more information, refer to https://www.elastic.co/support/eol[Elastic's Product End of Life Dates]. 

[float]
=== Plan for your upgrade

Before upgrading to {elastic-sec} 8.x, consider the following recommendations:

* Plan for an appropriate amount of time to complete the upgrade. Depending on your configuration and the size of your cluster, the process can take up to a week to complete.

> * Open a https://support.elastic.co[support case] with Elastic to alert our Elastic Support team of your system change. If your support subscription includes consultative support, you can also purchase a https://www.elastic.co/consulting[consulting engagement]. 

Reviewer: `System change` as in the whole entire upgrade is a big system change. Thanks for checking!

> * Open a https://support.elastic.co[support case] with Elastic to alert our Elastic Support team of your system change. If you need additional assistance beyond your support subscription level, you can also purchase a https://www.elastic.co/consulting[consulting engagement]. 

Reviewer 6: ```suggestion
* Open a https://support.elastic.co/[support case] with Elastic to alert our Elastic Support team of your system change. If you need additional assistance, https://www.elastic.co/consulting[Elastic Consulting Services] provides the technical expertise and step-by-step approach for upgrading your Elastic deployment.
```

* If your organization sends alerts (signals) to an external SOAR (security orchestration, automation, and response) platform, you may need to change your workflows to accommodate the new <<alert-schema, alert schema>> used in 8.x.

* Review release notes, deprecations, and breaking changes for {security-guide}/release-notes.html[{elastic-sec}], {ref}/es-release-notes.html[{es}], {kibana-ref}/release-notes.html[{kib}], and, if applicable, {fleet-guide}/release-notes.html[{fleet} and {agent}], {beats-ref}/release-notes.html[{beats}], and {logstash-ref}/releasenotes.html[{ls}]. Identify any issues that might affect your deployment. Work with your Support Engineer on any questions you may have. Start with breaking changes for your solution and platform components,such as {es} and {kib}. 

* Schedule a system maintenance window within your organization.

[float]
=== Pre-upgrade steps

To prepare for the upgrade process, follow these steps before you start:

. Do a software version inventory across your entire Elastic deployment, including {es}, {kib}, {agent}, {beats}, and {ls}. 

> . If you're running any of these versions earlier than 7.17, you must first upgrade them all to the latest 7.17.x patch release before upgrading to 8.x. This enables you to use the Upgrade Assistant to upgrade to 8.x. 

Reviewer 7: What does "these versions" refer to? Should there be a link here or a list after the step?

Reviewer: Revised to say "running any versions earlier than 7.17..."

+
NOTE: If you're managing your deployments using {ece-ref}/ece-upgrade.html[{ece}] (ECE) or {eck-ref}/k8s-upgrading-eck.html[{eck}] (ECK), you may need to upgrade the system clusters first.

> . Note that alerts in 8.x are written using the `kibana_system` user instead of the `current user`, so any ingest pipelines configured against the alert index will use the `kibana_system` user privileges, which may break your ingest pipeline.

Reviewer 4: This is good to add, since we know of at least customer that ran into it, but modifying the ingest pipelines for the alerts indices is not normal and not recommended.  Question: Should we add a parenthetical "not common" to avoid confusing users who don't know what we're talking about?   (ref: TurboTax does this when it asks you some arcane questions about special tax conditions that you probably don't have, and by saying "not common" it makes you feel more comfortable about skipping that section)

** If you're upgrading a self-orchestrated deployment, refer to {stack-ref}/upgrading-elasticsearch.html[these instructions] and upgrade the data nodes tier by tier in this order:
... Frozen tier
... Cold tier 
... Warm tier
... Hot tier 
... Any other nodes not in a tier
... All remaining nodes that are neither master-eligible nor data nodes
... Master-eligible nodes

. Upgrade {kib}. Refer to {stack-ref}/upgrading-kibana.html[these instructions].
+
NOTE: If you're using Elastic Cloud Hosted or {ece}, this is already included in the {es} upgrade.

. Validate that {es} and {kib} are operating as expected by completing the following checks: 
.. For {es}:
... Check the status of your clusters and ensure that they're green by running a `GET _cat/health` API request. For more information, refer to the {ref}/cat-health.html[cat health API documentation].
... Ensure that the index and search rate are close to what they were before upgrading. Go to **Stack Monitoring** -> **{es}** -> **Overview**.
+
TIP: You can also check the index document count using the {ref}/cat-indices.html[cat index API].

> ... Verify that {slm} (SLM) is taking snapshots by {ref}/snapshots-take-snapshot.html#check-slm-history[checking the SLM history]. 

Reviewer 7: Already defined earlier on this page. 

```suggestion
... Verify that SLM is taking snapshots by {ref}/snapshots-take-snapshot.html#check-slm-history[checking the SLM history]. 
```

> ... If you use {ml}, ensure that it is up and running. 

Reviewer 7: Should there be a link to instructions for how to do this?

Reviewer: @szabosteve any recommended ML links that would provide this? Maybe this link? https://www.elastic.co/guide/en/machine-learning/current/ml-ad-finding-anomalies.html

.. For {kib}: 
... Ensure that you and your users can successfully log in to {kib} and access desired pages.
... Check {kibana-ref}/discover.html[Discover] and verify that the index patterns you typically use are available.
... Verify that your commonly used {kibana-ref}/dashboard.html[dashboards] are available and working properly.
... If you use any Watcher-based {kib} scheduled {kibana-ref}/reporting-getting-started.html[reporting], ensure that it's working properly.

. Upgrade your ingest components (such as {ls}, {fleet} and {agent}, {beats}, etc.). For details, refer to the {stack-ref}/upgrading-elastic-stack.html[Elastic Stack upgrade docs].

> . Validate that Ingest is operating OK.

Reviewer 7: Using "OK" in place of "correctly" or "successfully" seems a bit too informal for our tech docs. Another option is "as expected".

```suggestion
. Validate that Ingest is operating correctly.
```

.. Open *Discover*, go through data views for each of your expected ingest data streams, and ensure that data is being ingested in the expected format and volume. 

> . Validate that {elastic-sec} is operating OK.

Reviewer 7: ```suggestion
. Validate that {elastic-sec} is operating correctly.
```

.. Re-enable your desired SIEM detection rules (rule management), and ensure that enabled rules are running without errors or warnings (rule monitoring).
.. Ensure that any SOAR workflows that consume alerts are working.
.. Verify that any custom dashboards your team has created are working properly, especially if they operate on alert (signal) documents.

> . If you performed these steps on a non-prod deployment, repeat these same steps on your prod environment. If you're using a cross-cluster search environment and performed these steps on your remote clusters, repeat these same steps on your other deployments. 

Reviewer 7: ```suggestion
. If you performed these steps on a non-production deployment, repeat these same steps on your production environment. If you're using a cross-cluster search environment and performed these steps on your remote clusters, repeat these same steps on your other deployments. 
```

. Click *Select all _x_ rules*, or individually select the rules you want to re-enable.
. Click *Bulk actions -> Enable* to re-enable the rules.

Alternatively, you can use the <<bulk-actions-rules-api, Bulk rule actions>> API to re-enable rules.

[float]
[[fda-upgrade]]
==== Full Disk Access (FDA) approval for {elastic-endpoint}

When you manually install {elastic-endpoint}, you must approve a system extension, kernel extension, and enable Full Disk Access (FDA). There is a new FDA requirement in 8.x. Refer to <<elastic-endpoint-deploy-reqs>> to review the required permissions.

[float]
[[data-views-upgrade]]
==== Requirements to display Data views in the {security-app}

To make the *Data view* option appear in an environment with legacy alerts, a user with elevated role privileges must visit the {security-app}, open a page that displays alert data (at least one alert must be present), and then refresh the page. The user's role privileges must allow them to enable the detections feature in a {kib} space. For more information, refer to <<enable-detections-ui, Enable and access detections>>. 

[float]
[[alert-schema-upgrade]]
==== New alert schema

> The system index for detection alerts has been renamed from `.siem-signals-<space-id>` to `.alerts-security.alerts-<space-id>` and is now a hidden index. Therefore, the schema used for alert documents in {elastic-sec} has changed. Users that access documents in the `.siem-signals` indices using the {elastic-sec} API must modify their API queries and scripts to operate properly on the new 8.x alert documents. Refer to <<query-alert-indices, how to query alert indices>> and review the new <<alert-schema, Alert schema>>.

Reviewer 7: [Here](https://github.com/elastic/security-docs/pull/5186/files#diff-c6f4f6d020765a7b8e401186e3a0da3d43ba14059b4a20cfebd5765f44f36b9fR101), "signals" is added in parentheses after "alert". Do you need to do the same in this paragraph for consistency?

Reviewer 7: I could be misremembering, but I vaguely recall "signals" being used synonymously with "alerts" in 7.x and even sometimes in 8.x (e.g., the `max_signals` field allows you to define the max number of alerts a rule can generate). That said, it's probably not necessary to make the change that I suggested. It doesn't add any clarity or necessary context to the explanation about the new alert schema. Feel free to reject this suggestion!

Reviewer 4: We've attempted to avoid/eliminate the use of "signals" altogether in 8.x.  (@nastasha-solomon 's comment about the max_signals field notwithstanding), so can we review all uses of "(signals)" to ensure that we're not persisting its use in 8.x?  Perhaps "(signals in 7.x)" would be more descriptive, or "(formerly signals)" or something similar.

[float]
[[preview-upgrade]]
==== New privileges required to view alerts and preview rules

> * To view alerts, users need `manage`, `write`, `read`, and `view_index_metadata` privileges to two new indices, `.alerts-security.alerts` and `.internal.alerts-security.alerts`. Existing users who are upgrading to 8.x can retain their privileges to the `.siem-signals` index.

Reviewer 7: ```suggestion
* To view alerts, users need `manage`, `write`, `read`, and `view_index_metadata` privileges for two new indices, `.alerts-security.alerts` and `.internal.alerts-security.alerts`. Existing users who are upgrading to 8.x can retain their privileges to the `.siem-signals` index.
```

* To <<preview-rules, preview rules>>, users need `read` access to the new `.preview.alerts-security.alerts` index. Refer to <<detections-permissions-section>> for more information.

[float]
[[im-rules-upgrade]]

> ==== Updates to indictor match rules

Reviewer 4: "indictor" should be "indicator"
